cmake/extensions.cmake

```
@@ -3377,9 +3377,14 @@ function(zephyr_linker_dts_memory)
 
   dt_reg_addr(addr PATH ${DTS_MEMORY_PATH})
   dt_reg_size(size PATH ${DTS_MEMORY_PATH})
+  dt_prop(name PATH ${DTS_MEMORY_PATH} PROPERTY "memory-region")
+  if (NOT DEFINED name)
+    # Fallback to the node path
+    set(name ${DTS_MEMORY_PATH})
+  endif()
```

minor nit, it's possible to do:
```suggestion
set_ifndef(name ${DTS_MEMORY_PATH})
```

https://github.com/zephyrproject-rtos/zephyr/blob/92fbd7da14d1c0c69457441fb13838d0b81f7f12/cmake/extensions.cmake#L1619